apps/experiments/tasks.py

```diff
@@ -1,20 +1,42 @@
+import io
 import time
-from datetime import datetime
+from datetime import timedelta
 
 from celery.app import shared_task
+from django.core.files.base import ContentFile
+from django.utils import timezone
 from langchain_core.messages import AIMessage, HumanMessage
 from taskbadger.celery import Task as TaskbadgerTask
 
 from apps.channels.datamodels import Attachment, BaseMessage
 from apps.chat.bots import create_conversation
 from apps.chat.channels import WebChannel
+from apps.experiments.export import experiment_to_csv
 from apps.experiments.models import Experiment, ExperimentSession, PromptBuilderHistory, SourceMaterial
+from apps.files.models import File
 from apps.service_providers.models import LlmProvider, LlmProviderModel
 from apps.teams.utils import current_team
 from apps.users.models import CustomUser
 from apps.utils.taskbadger import update_taskbadger_data
 
 
+@shared_task(bind=True, base=TaskbadgerTask)
+def async_export_chat(self, experiment_id: int, tags: list[str] = None, participant: str = None) -> dict:
+    experiment = Experiment.objects.get(id=experiment_id)
+    csv_in_memory = experiment_to_csv(experiment, tags, participant)
+    bytes_buffer = io.BytesIO(csv_in_memory.getvalue().encode("utf-8"))
+    bytes_buffer.seek(0)
+    uploaded_file = ContentFile(content=bytes_buffer.read(), name="chat_export.csv")
```

Comment:
nit: BytesIO is not needed

```suggestion
    uploaded_file = ContentFile(content=csv_in_memory.getvalue().encode("utf-8"), name="chat_export.csv")
```